cool-demos/conditional-blocks.html

```diff
@@ -0,0 +1,452 @@
+            win.customElements.define("conditional-block", ConditionalBlock);
+        }
+
+        // Custom function that manages the Insert/edit dialog
+        const dialogManager = (conditionalBlock, editor) => {
+            // Open a TinyMCE modal where the user can set the badge's
+            // background and text color.
+            // https://www.tiny.cloud/docs/ui-components/dialog/
+            // https://www.tiny.cloud/docs/ui-components/dialogcomponents/
+            editor.windowManager.open({
+                title: 'Insert/edit Conditional block',
+                body: {
+                    type: 'panel',
+                    items: [
+                        {
+                            type: 'selectbox',
+                            name: 'property',
+                            label: 'Property',
+                            items: [
+                                { value: 'number_of_people', text: 'number_of_people' },
+                                { value: 'name_of_event', text: 'name_of_event' },
+                                { value: 'length_of_stay', text: 'length_of_stay' },
+                            ]
+                        }, {
+                            type: 'selectbox',
+                            name: 'operator',
+                            label: 'Operator',
+                            items: [
+                                { value: 'is greater than', text: 'is greater than' },
+                                { value: 'is equal or greater than', text: 'is equal or greater than' },
+                                { value: 'is equal to', text: 'is equal to' },
+                                { value: 'is equal or less than', text: 'is equal or less than' },
+                                { value: 'is less than', text: 'is less than' },
+                                { value: 'is not equal to', text: 'is not equal to' },
+                            ]
+                        }, {
+                            type: 'input',
+                            name: 'value',
+                            label: 'Value'
+                        }
+                    ]
+                },
+                buttons: [
+                {
+                    type: 'cancel',
+                    name: 'closeButton',
+                    text: 'Cancel'
+                },
+                {
+                    type: 'submit',
+                    name: 'submitButton',
+                    text: 'Save',
+                    primary: true
+                }
+                ],
+                initialData: {
+                    property: conditionalBlock ? conditionalBlock.dataset.property : 'null' ,
+                    operator: conditionalBlock ? conditionalBlock.dataset.operator : 'is equal to',
+                    value: conditionalBlock ? conditionalBlock.dataset.value: '0'
+                },
+                onSubmit: function (dialog) {
```

Comment:
Why use `function` here when elsewhere arrow functions are used?

Reply:
Copy and paste blindness